tests/phpunit/Civi/Api4/Action/CreditNoteAllocation/AllocateActionTest.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+use Civi\Api4\CreditNote;
+use Civi\Api4\CreditNoteAllocation;
+use Civi\Financeextras\Test\Helper\CreditNoteTrait;
+use Civi\Financeextras\Utils\FinancialAccountUtils;
+
+/**
+ * CreditNote.AllocateAction API Test Case.
+ *
+ * @group headless
+ */
+class Civi_Api4_CreditNoteAllocation_AllocateActionTest extends BaseHeadlessTest {
+
+  use CreditNoteTrait;
+
+  /**
+   * Test credit note compute action returns expected fields.
+   */
+  public function testCanAllocateCreditNoteToContribution() {
+    $amountAllocated = 100;
+    $contributionAmount = 200;
+    $creditNote = $this->createCreditNote();
+    $contribution = $this->createContribution($creditNote['contact_id'], $contributionAmount);
+    $allocation = $this->createAllocation($creditNote['id'], $contribution['id'], $amountAllocated, $creditNote['currency']);
+
+    $this->assertNotEmpty($allocation);
+    $this->assertEquals(CRM_Core_BAO_FinancialTrxn::getTotalPayments($contribution['id'], TRUE), $contributionAmount - $amountAllocated);
+  }
+
+  /**
+   * Tests that the expected accounting entries are created for the credit note allocation.
+   */
+  public function testExpectedAccountingEntriesAreCreatedForCrediNoteAllocation() {
+    $amountAllocated = 100;
+    $creditNote = $this->createCreditNote();
+    $contribution = $this->createContribution($creditNote['contact_id'], 200);
+    $this->createAllocation($creditNote['id'], $contribution['id'], $amountAllocated, $creditNote['currency']);
+
+    $expectedAccount = FinancialAccountUtils::getFinancialTypeAccount(
+        $creditNote['items'][0]['financial_type_id'],
+        'Accounts Receivable Account is'
+      );
+
+    $entityFinancialTrxn = \Civi\Api4\EntityFinancialTrxn::get(FALSE)
+      ->addWhere('entity_id', '=', $contribution['id'])
+      ->addWhere('entity_table', '=', 'civicrm_contribution')
+      ->addWhere('amount', '=', $amountAllocated)
+      ->execute()
+      ->first();
+
+    $this->assertNotEmpty($entityFinancialTrxn);
+
+    $financialTrxn = \Civi\Api4\FinancialTrxn::get(FALSE)
+      ->addWhere('from_financial_account_id', '=', $expectedAccount)
+      ->addWhere('to_financial_account_id', '=', $expectedAccount)
+      ->addWhere('total_amount', '=', $amountAllocated)
+      ->addWhere('id', '=', $entityFinancialTrxn['financial_trxn_id'])
+      ->addWhere('status_id', '=', 1)
+      ->addWhere('payment_processor_id', 'IS NULL')
+      ->addWhere('payment_instrument_id', '=', 1)
+      ->addWhere('check_number', 'IS NULL')
+      ->execute()
+      ->first();
+    $this->assertNotEmpty($financialTrxn);
```

Comment:
You also need to add an assertion to check if an entity_financial_trxn is created for the line item.


![image](https://github.com/compucorp/io.compuco.financeextras/assets/6275540/312cd65a-7d6e-4cb5-962f-e4c5f81bb800)